# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/PerformanceObserver.webidl
# Commit: 5248abfed275
# Full Hash: 5248abfed275b2938ed64770bec58b3ff68d624b
# Author: Will Hawkins <whawkins@mozilla.com>
# Date: 2019-04-12 21:56:33
# Regressor Bug: 1539006
# File Overlap Count: 1
# Description:
#   Bug 1539006: Properly implement Performance Timeline Level 2 w3c spec. r=mstange,baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27008
# ==============================================================================

diff -r e17500911824 -r 5248abfed275 dom/webidl/PerformanceObserver.webidl
--- a/dom/webidl/PerformanceObserver.webidl	Fri Apr 12 15:49:22 2019 +0000
+++ b/dom/webidl/PerformanceObserver.webidl	Fri Apr 12 16:29:48 2019 +0000
@@ -8,8 +8,9 @@
  */
 
 dictionary PerformanceObserverInit {
-  required sequence<DOMString> entryTypes;
-  boolean buffered = false;
+  sequence<DOMString> entryTypes;
+	DOMString type;
+  boolean buffered;
 };
 
 callback PerformanceObserverCallback = void (PerformanceObserverEntryList entries,
@@ -19,7 +20,8 @@
  Constructor(PerformanceObserverCallback callback),
  Exposed=(Window,Worker)]
 interface PerformanceObserver {
-    void observe(PerformanceObserverInit options);
+    [Throws] void observe(optional PerformanceObserverInit options);
     void disconnect();
     PerformanceEntryList takeRecords();
+    static readonly attribute object supportedEntryTypes;
 };